x-pack/legacy/plugins/apm/server/routes/create_api.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { merge } from 'lodash';
+import Boom from 'boom';
+import { InternalCoreSetup } from 'src/core/server';
+import { Request } from 'hapi';
+import * as t from 'io-ts';
+import { PathReporter } from 'io-ts/lib/PathReporter';
+import { ServerAPI, RouteFactoryFunction, RouteParams } from './typings';
+import { removeUndefinedProps } from '../../public/context/UrlParamsContext/helpers';
+
+type GenericRouteFactoryFunction = RouteFactoryFunction<
+  any,
+  any,
+  { [key in keyof RouteParams]?: RouteParams[key] },
+  any
+>;
+
+export function createApi() {
+  const factoryFns: GenericRouteFactoryFunction[] = [];
+  const router = {
+    add(fn: GenericRouteFactoryFunction) {
+      factoryFns.push(fn);
+      return router;
+    },
+    init: (core: InternalCoreSetup) => {
+      const { server } = core.http;
+      factoryFns.forEach(fn => {
+        const { params = {}, ...route } = fn(core);
+        server.route(
+          merge(
+            {
+              options: {
+                tags: ['access:apm']
+              }
```

Comment:
Do we want this tag on all routes? Maybe we do, but it feels like a fairly specific think to bake into a generic router.

Reply:
It's on all routes, except for the breakdown metrics, and that's simply because I forgot about them. So looks like it's a fine default for now?